### ISSUE TITLE
Update nuxt

### ISSUE THREAD

**Issue by antonioortegajr** [csjoblom/hackforacause-website#100]
The version of nuxt on the site is outdated and needs an updating to the latest version.

**Comment by websupergirl** [comment]
Updating nuxt to v2.10.2 causes an eslint-loader error.

Updating eslint-loader to v2.1.2 causes a new compile error, so here I am:

 ```
ERROR  Failed to compile with 3 errors               friendly-errors 22:46:06

These relative modules were not found:                friendly-errors 22:46:06
                                                      friendly-errors 22:46:06
* ../../../../../../assets/images/bg2.jpg in ./node_modules/css-loader/dist/cjs.js??ref--7-oneOf-1-1!./node_modules/postcss-loader/src??ref--7-oneOf-1-2!./node_modules/sass-loader/lib/loader.js??ref--7-oneOf-1-3!./assets/scss/hack.scss
* ../../../../../../assets/images/faq.jpg in ./node_modules/css-loader/dist/cjs.js??ref--7-oneOf-1-1!./node_modules/postcss-loader/src??ref--7-oneOf-1-2!./node_modules/sass-loader/lib/loader.js??ref--7-oneOf-1-3!./assets/scss/hack.scss
* ../../../../../../assets/images/submit-a-challenge-bg.jpg in ./node_modules/css-loader/dist/cjs.js??ref--7-oneOf-1-1!./node_modules/postcss-loader/src??ref--7-oneOf-1-2!./node_modules/sass-loader/lib/loader.js??ref--7-oneOf-1-3!./assets/scss/hack.scss

```

**Comment by websupergirl** [comment]
Links in scss files have to be `~@/assets/` not `~/assets/` ... this fixes the issue above.

**Comment by websupergirl** [comment]
So that was fixed but this is not really linting on save as Chris thought it was, because ... this (in pretty much every file, it won't run dev.)

![Screen Shot 2019-10-22 at 11 15 14 PM](https://user-images.githubusercontent.com/38568655/67363304-ee2a8280-f521-11e9-874a-b57335bd9c4e.png)

**Comment by csjoblom** [comment]
Ah it just runs the lint it appears, if you use the lintfix npm script you should stop seeing that error.

**Comment by websupergirl** [comment]
I can't get it to fix with npm or yarn or eslint or whatever lol

If you mean `npm run lint -- --fix` then it just keeps giving me the same errors and not fixing.
```
✖ 951 problems (902 errors, 49 warnings)
  901 errors and 40 warnings potentially fixable with the --fix option.
```

Also trying to upgrade the eslint to v6 gives you this:
`vue-eslint-parser@5.0.0 requires a peer of eslint@^5.0.0 but none is installed. You must install peer dependencies yourself.`
which apparently is a known issue:
https://github.com/vuejs/eslint-plugin-vue/issues/920

**Comment by csjoblom** [comment]
npm run lintfix is just a package.json script that runs eslint with the appropriate options. Do you have eslint installed globally? Could be a version issue and I would recommend using the version eslint (installed locally)

**Comment by csjoblom** [comment]
I would recommend opening the package.json and running the eslint command manually from the project root directory. Make sure your eslint is the appropriate version beforehand by doing eslint —version

**Comment by csjoblom** [comment]
The version of eslint that comes with this repo is 4.12 and your error log has version 5.0 so this issue is likely also related to your local pathing/install.

**Comment by csjoblom** [comment]
![FE73C807-C9E0-4EAC-902A-DC38AFDDB947](https://user-images.githubusercontent.com/3222182/67401893-9f520c80-f564-11e9-8a32-36a0a97a3eeb.jpeg)

**Comment by csjoblom** [comment]
Does the npm run lintfix command work?

**Comment by websupergirl** [comment]
I saw your comment (somewhere?) about the version so to confirm, I have tried it with the version included here, with the updated version of 5, with the version of 6 ... it doesn't work, no. It runs, but it errors out with lint --fix or lintfix.

Right now it's running
```
├── babel-eslint@8.2.6
├── bootstrap@4.3.1
├── cross-env@5.2.1
├── eslint@4.19.1
├── eslint-config-prettier@2.10.0
├── eslint-friendly-formatter@3.0.0
├── eslint-loader@2.2.1
├── eslint-plugin-prettier@2.7.0
├── eslint-plugin-vue@4.7.1
├── hamburgers@0.9.3
├── UNMET PEER DEPENDENCY jquery@1.9.1 - 3
├── node-sass@4.12.0
├── npm@5.10.0
├── nuxt@2.10.2
├── UNMET PEER DEPENDENCY popper.js@^1.14.7
├── prettier@1.16.3
├── push-dir@0.4.1
├── sass-loader@6.0.7
└── vue-awesome-swiper@3.1.3
```

**Comment by websupergirl** [comment]
Just FYI these prettier spacing errors weren't there on base download. They just showed up blocking run after the nuxt / eslint-loader update.

**Comment by csjoblom** [comment]
I would recommend getting a clean repo with the stated versions. I wouldn’t recommend any upgrades unless you plan on upgrading the configs as well and dealing with any other dependency issues. Lintfix is running fine for me and based on your statement above lint was correctly finding issues. If you can’t get lintfix to run with the repo dependency versions then we may need to take this offline as it has been functioning for a long time without issue and the linter is still working just fine I’m
builds.

**Comment by csjoblom** [comment]
I would recommend not upgrading any npm packages until you have a little more experience with node and fixing dependency issues.

**Comment by csjoblom** [comment]
![image](https://user-images.githubusercontent.com/3222182/67405810-1f2ea580-f56a-11e9-90ec-40c3a9d26916.png)

Fresh repo, lint and lintfix are both running correctly.

**Comment by csjoblom** [comment]
https://nuxtjs.org/guide/upgrading/ is nuxt's upgrade guide, if eslint is upgraded there is likely a guide for that as well on their site.

**Comment by websupergirl** [comment]
Did you update nuxt?

**Comment by csjoblom** [comment]
Nuxt is just a bunch of npm packages glued together, I haven't upgraded as it hasn't been necessary other than to keep up with tech debt. I can take a crack at upgrading everything if you'd like and then have you review the changes.

**Comment by csjoblom** [comment]
You can see the nuxt dependencies in node_modules > nuxt > package.json

**Comment by csjoblom** [comment]
Any update will likely need to have configs touched, scripts changed, and etc. Unfortunately that's just how it goes, it takes a little bit of research for changes on any updated packages.

**Comment by csjoblom** [comment]
A better starter ticket may be #104 as its just a single dependency swap.

**Comment by websupergirl** [comment]
Just FYI the nuxt upgrade is required to upgrade to the bootstrap vue package and the nuxt-fontawesome. That is why the issue exists. And yes fixing the dependency issues and researching them all is what I was doing - that is why the information above exists (the loader fix, the issues with eslint vue, etc). I also read all the upgrade guides which is how I tested this on varying versions of only upgrading a few things, upgrading all, upgrading partial (like bringing up everything to eslint 5 era), and so on. I'm just not finding why the base + nuxt/loader upgrade is finding spacing issues that didn't exist before.

**Comment by csjoblom** [comment]
I was able to upgrade to nuxt 2.10 without any other dependency changes, I'll submit a PR shortly. As far as updating the other dependencies those should be opened as separate issues for documentation purposes.

**Comment by websupergirl** [comment]
You did not get the loader issue on `npm run dev` after nuxt upgrade?

**Comment by csjoblom** [comment]
nuxt.config.js required a minor tweak outlined in the 2.0 migration blog post:

> Remove deprecated features from Nuxt 1.0 : Removed context.isServer and context.isClient (Use process.client and process.server)

I'm applying the asset pipeline fix as well with my PR.

**Comment by csjoblom** [comment]
> Warning: Starting from Nuxt 2.0 the ~/ alias won't be resolved correctly in your CSS files. You must use ~assets (without a slash) or the @ alias in url CSS references, i.e. background: url("~assets/banner.svg")

**Comment by websupergirl** [comment]
Yeah that was the one I fixed in my own local above.

**Comment by csjoblom** [comment]
@websupergirl check out my branch (upgrade-nuxt-latest) and see if it runs for you. I've created a PR #106

**Comment by websupergirl** [comment]
That one seems to be loading properly.

**Comment by csjoblom** [comment]
Fixed in #106
